examples/advanced/demo_optimizing_physics_parameter.py

```diff
@@ -0,0 +1,258 @@
+# We can use a simple optimization algorithm - Projected Gradient Descent - to enforce this constraint.
+# The following function allows one to compute the orthogonal projection onto the simplex, by a sorting algorithm (Reference: Large-scale Multiclass Support Vector Machine Training via Euclidean Projection onto the Simplex
+# Mathieu Blondel, Akinori Fujino, and Naonori Ueda)
+#
+
+
+@torch.no_grad()
+def projection_simplex_sort(v):
+    shape = v.shape
+    B = shape[0]
+    v = v.view(B, -1)
+    n_features = v.size(1)
+    u = torch.sort(v, descending=True, dim=-1).values
+    cssv = torch.cumsum(u, dim=-1) - 1.0
+    ind = torch.arange(n_features, device=v.device)[None, :].expand(B, -1) + 1.0
+    cond = u - cssv / ind > 0
+    rho = ind[cond][-1]
+    theta = cssv[cond][-1] / rho
+    w = torch.maximum(v - theta, torch.zeros_like(v))
+    return w.reshape(shape)
+
+
+# We also define a data fidelity term
+data_fidelity = dinv.optim.L2()
+# %%
+#
+# Run the algorithm
+#
+# Initialize a constant kernel
+kernel_init = torch.zeros_like(true_kernel)
+kernel_init[..., 5:-5, 5:-5] = 1.0
+kernel_init = projection_simplex_sort(kernel_init)
+n_iter = 1000
+stepsize = 0.7
+
+kernel_hat = kernel_init
+losses = []
+for i in tqdm(range(n_iter)):
+    # compute the gradient
+    with torch.enable_grad():
+        kernel_hat.requires_grad_(True)
+        physics.update_parameters(filter=kernel_hat)
+        loss = data_fidelity(y=y, x=x, physics=physics) / y.numel()
+        loss.backward()
+    grad = kernel_hat.grad
+
+    # gradient step and projection step
+    with torch.no_grad():
+        kernel_hat = kernel_hat - stepsize * grad
+        kernel_hat = projection_simplex_sort(kernel_hat)
+
+    losses.append(loss.item())
+
+dinv.utils.plot(
+    [true_kernel, kernel_init, kernel_hat],
+    titles=["True kernel", "Init. kernel", "Estimated kernel"],
+    suptitle="Result with Projected Gradient Descent",
+)
+
+# %%
+#
+# We can plot the loss to make sure that it decreases
+#
+plt.figure()
+plt.plot(range(n_iter), losses)
+plt.title("Loss evolution")
+plt.yscale("log")
+plt.xlabel("Iteration")
+plt.tight_layout()
+plt.show()
+
+# %%
+# Combine with arbitrary optimizer
+# --------------------------------
+#
+# Pytorch provides a wide range of optimizer for training neural networks.
+# We can also pick one of those to optimizer our parameter
+
+kernel_init = torch.zeros_like(true_kernel)
+kernel_init[..., 5:-5, 5:-5] = 1.0
+kernel_init = projection_simplex_sort(kernel_init)
+
+kernel_hat = kernel_init.clone()
+optimizer = torch.optim.Adam([kernel_hat], lr=0.1)
+
+# We will alternate a gradient step and a projection step
+losses = []
+n_iter = 200
+for i in tqdm(range(n_iter)):
+    optimizer.zero_grad()
+    # compute the gradient, this will directly change the gradient of `kernel_hat`
+    with torch.enable_grad():
+        kernel_hat.requires_grad_(True)
+        physics.update_parameters(filter=kernel_hat)
```

Comment:
we should use 
```suggestion
        physics.update(filter=kernel_hat)
```

(update_parameters shouldn't be called explicitly, as it is handled internally by `update`)

Reply:
I'm not fully understand the logic of `update` and `update_parameters`, can you clarify? Currently I suppose that they do the same job.

Reply:
physics.update() calls internally the update_parameters() of both the operator **and the noise**.

So if you do physics.update(sigma=.1, filter=filter) this will call


physics.update_parametera(filter=filter)
physics.noise_model.update_parameters(sigma=sigma)